packages/did/src/didAuthorizeExtrinsics/didAuthorizeExtrinsics.spec.ts

```diff
@@ -0,0 +1,170 @@
+          authentication: new Set([fullDid.authenticationKey.id]),
+        },
+      })
+      const extrinsic = await mockApi.tx.ctype.add('test-ctype')
+
+      await expect(async () =>
+        didAuthorizeExtrinsics({
+          did: newFullDid,
+          batchFunction: mockApi.tx.utility.batchAll,
+          extrinsics: [extrinsic, extrinsic],
+          nonce: new BN(0),
+          sign,
+          submitter: keypair.address,
+        })
+      ).rejects.toMatchInlineSnapshot(
+        '[DidBuilderError: Found no key for relationship "assertionMethod"]'
+      )
+    })
+
+    it('adds different batches requiring different keys', async () => {
+      const ctype1Extrinsic = await mockApi.tx.ctype.add(randomAsHex(32))
+      const ctype2Extrinsic = await mockApi.tx.ctype.add(randomAsHex(32))
+      const delegationExtrinsic1 = await mockApi.tx.delegation.createHierarchy(
+        randomAsHex(32),
+        randomAsHex(32)
+      )
+      const delegationExtrinsic2 = await mockApi.tx.delegation.createHierarchy(
+        randomAsHex(32),
+        randomAsHex(32)
+      )
+      const ctype3Extrinsic = await mockApi.tx.ctype.add(randomAsHex(32))
+      const ctype4Extrinsic = await mockApi.tx.ctype.add(randomAsHex(32))
+
+      const batchFunction =
+        jest.fn() as unknown as typeof mockApi.tx.utility.batchAll
+      const extrinsics = [
+        ctype1Extrinsic,
+        ctype2Extrinsic,
+        delegationExtrinsic1,
+        delegationExtrinsic2,
+        ctype3Extrinsic,
+        ctype4Extrinsic,
+      ]
+      await didAuthorizeExtrinsics({
+        did: fullDid,
+        batchFunction,
+        extrinsics,
+        nonce: new BN(0),
+        sign,
+        submitter: keypair.address,
+      })
+
+      expect(batchFunction).toHaveBeenCalledWith([
+        ctype1Extrinsic,
+        ctype2Extrinsic,
+      ])
+      expect(batchFunction).toHaveBeenCalledWith([
+        delegationExtrinsic1,
+        delegationExtrinsic2,
+      ])
+      expect(batchFunction).toHaveBeenCalledWith([
+        ctype3Extrinsic,
+        ctype4Extrinsic,
+      ])
+    })
+  })
+
+  // TODO: complete these tests once SDK has been refactored to work with generic api object
+  describe('.build()', () => {
+    it('throws if batch is empty', async () => {
+      await expect(async () =>
+        didAuthorizeExtrinsics({
+          did: fullDid,
+          batchFunction: mockApi.tx.utility.batchAll,
+          extrinsics: [],
+          sign,
+          submitter: keypair.address,
+        })
+      ).rejects.toMatchInlineSnapshot(
+        '[DidBuilderError: Cannot build a batch with no transactions]'
+      )
+    })
+    it.todo('successfully create a batch with only 1 extrinsic')
+    it.todo('successfully create a batch with 1 extrinsic per required key')
+    it.todo('successfully create a batch with 2 extrinsics per required key')
+    it.todo(
+      'successfully create a batch with 1 extrinsic per required key, repeated two times'
+    )
+  })
+})
```

Comment:
What do you think about tests, with changing did keys?

Reply:
Again, this was carried over from https://github.com/KILTprotocol/sdk-js/blob/013633a/packages/did/src/DidBatcher/DidBatchBuilder.spec.ts#L135-L148